client/server/render/index.js

```diff
@@ -125,6 +125,22 @@ export function render( element, key = JSON.stringify( element ), req ) {
 	} catch ( ex ) {
 		if ( process.env.NODE_ENV === 'development' ) {
 			throw ex;
+		} else {
+			try {
+				req.context.store.dispatch(
```

Comment:
I think this Redux overhead is pretty much useless and outdated for such a simple purpose like logging stuff to Logstash. Especially in server-only code.

I'm not sure if `req.context.store` is guaranteed to be always available.

Rewriting this to a `superagent` HTTP request would be preferrable:
```js
request.post( 'https://public-api.wordpress.com/rest/v1.1/logstash ).send( { ... } ).end();
```

Server-side `wpcom` requests don't sent auth cookies, if I'm not mistaken, not even with user bootstrapping on. So the `wpcom` lib doesn't provide any additional value here.

(this is all optional and more suitable for a followup PR, rather than blocking this one)

Reply:
+1

If we are going to revisit how to capture logs, I'd first explore the option to just output stuff to stdout (or a file) and raise a request to systems to capture it and send it to logstash. 

Only if that option is not possible I'd go with sending a request to /logstash. Fully agree that  using `superagent` or equivalent is better than going via Redux.

Reply:
The `/logstash` endpoint is already available, stable, and used by several other features. So I don't see the upside of reimplementing the logging from scratch using another method. What would be the benefit of logging the `calypso_ssr` errors to stderr?

Reply:
Is mostly about decomposition: not having a dependency on `/logstash` API service, reduce traffic on `public-api.wordpress.com` servers, not being limited to the payload schema and marshalling imposed by `/logstash`. It follows 12 factor apps pattern more closely.

Logging to the console will also improve the dev experience, as I can see in my console everything we would send to logstash.